### ISSUE TITLE
Error Report: App freeze after logging in with some student SSO credentials

### ISSUE THREAD

**Issue by c3bryant** [UCSD/campus-mobile#305]
Two student beta testers have reported that the 6.0 (3) iOS build is freezing immediately after they successfully login with their student SSO credentials. This bug is currently affecting 20% of iOS beta users.

Report 1: 
> The app is not working for me. I am able to open it, log in, but once I am logged in the, the screen is dark blue and then quits. I deleted the app, restarted my phone, and it still did not work. Now every time I open the app, it quits immediately.

Report 2:
> The app worked for one second however after I interacted with the app, it just showed a blue screen and will not let me do anything else, even when the app is refreshed.

**Comment by Gulci** [comment]
It has been found that multiple factors can play into this issue. So far two issues have been discovered:

* The AcademicHistory API can sometimes return `null` for the day of a class, which the app currently does not handle. While we wait on learning about more exceptions to the rule, we have simply prevented the app from crashing. We will later handle special cases specifically.
* There was an issue with the UI code for the FinalsCard which has been resolved.

This issue is ongoing as we learn more about the responses we can receive from the MyAcademicHistory API.

**Comment by c3bryant** [comment]
One new issue has arisen with an update to the My/AcademicHistory API QA. Sections without a `time` were previously returning a string, and now these sections will return `null`. This is resulting in the following err:
```
Cannot read property 'split' of null"
```
Issue origin is suspected to be in app/util/schedule

**Comment by Gulci** [comment]
Commit 20a410e resolves the `Cannot read property 'split' of null"` error. The UI has been modified appropriately for those cases.

![image](https://user-images.githubusercontent.com/8071536/41131398-74dfed76-6a70-11e8-9298-afd0335a007a.png)

**Comment by c3bryant** [comment]
All known cases of this issue happening have been resolved and verified in v6.0-rc, closing.
